pandas/io/json/_normalize.py

```diff
@@ -228,13 +228,17 @@ def json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js, spec):
-        result = js
+    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> List:
         if isinstance(spec, list):
+            _res = js
             for field in spec:
-                result = result[field]
+                _res = _res[field]
+            result = cast(List, _res)
         else:
-            result = result[spec]
+            result = js[spec]
+
+        if not isinstance(result, Iterable):
```

Comment:
this is even more strange

Reply:
if the tests don't fail then I am puzzled what exactly you are changing here

Reply:
This test fails (or are you talking about other tests?) when e.g. the nulls_fixtures is used. It will be a float then (float('NaN')). As a 'str' I also iterable (And valid according to the tests), I'm testing for this.

It was how I could make the existing tests pass and the newly added.

Reply:
Should this be if not `isinstance(result, list)` or should the return type of _pull_field be typed as Iterable?

Reply:
I’ll change the return type to `Iterable` (`str` is valid and is not a list)